src/HelmetUtils.js

```diff
@@ -0,0 +1,530 @@
+        }
+    }
+
+    for (let i = attributesToRemove.length - 1; i >= 0; i--) {
+        elementTag.removeAttribute(attributesToRemove[i]);
+    }
+
+    if (helmetAttributes.length === attributesToRemove.length) {
+        elementTag.removeAttribute(HELMET_ATTRIBUTE);
+    } else if (elementTag.getAttribute(HELMET_ATTRIBUTE) !== attributeKeys.join(",")) {
+        elementTag.setAttribute(HELMET_ATTRIBUTE, attributeKeys.join(","));
+    }
+};
+
+const updateTags = (type, tags) => {
+    const headElement = document.head || document.querySelector(TAG_NAMES.HEAD);
+    const tagNodes = headElement.querySelectorAll(`${type}[${HELMET_ATTRIBUTE}]`);
+    const oldTags = Array.prototype.slice.call(tagNodes);
+    const newTags = [];
+    let indexToDelete;
+
+    if (tags && tags.length) {
+        tags.forEach(tag => {
+            const newElement = document.createElement(type);
+
+            for (const attribute in tag) {
+                if (tag.hasOwnProperty(attribute)) {
+                    if (attribute === TAG_PROPERTIES.INNER_HTML) {
+                        newElement.innerHTML = tag.innerHTML;
+                    } else if (attribute === TAG_PROPERTIES.CSS_TEXT) {
+                        if (newElement.styleSheet) {
+                            newElement.styleSheet.cssText = tag.cssText;
+                        } else {
+                            newElement.appendChild(document.createTextNode(tag.cssText));
+                        }
+                    } else {
+                        const value = (typeof tag[attribute] === "undefined") ? "" : tag[attribute];
+                        newElement.setAttribute(attribute, value);
+                    }
+                }
+            }
+
+            newElement.setAttribute(HELMET_ATTRIBUTE, "true");
+
+            // Remove a duplicate tag from domTagstoRemove, so it isn't cleared.
+            if (oldTags.some((existingTag, index) => {
+                indexToDelete = index;
+                return newElement.isEqualNode(existingTag);
+            })) {
+                oldTags.splice(indexToDelete, 1);
+            } else {
+                newTags.push(newElement);
+            }
+        });
+    }
+
+    oldTags.forEach(tag => tag.parentNode.removeChild(tag));
+    newTags.forEach(tag => headElement.appendChild(tag));
+
+    return {
+        oldTags,
+        newTags
+    };
+};
+
+const generateElementAttributesAsString = (attributes) => Object.keys(attributes)
+    .reduce((str, key) => {
+        const attr = typeof attributes[key] !== "undefined"
+            ? `${key}="${attributes[key]}"`
+            : `${key}`;
+        return str ? `${str} ${attr}` : attr;
+    }, "");
+
+const generateTitleAsString = (type, title, attributes, encode) => {
+    const attributeString = generateElementAttributesAsString(attributes);
+    return attributeString
+        ? `<${type} ${HELMET_ATTRIBUTE}="true" ${attributeString}>${encodeSpecialCharacters(title, encode)}</${type}>`
```

Comment:
`encodeSpecialCharacters` - is this the "functional" way of doing it?  seems confusing, what if the check for `encode` was here so people don't think it's _always_ encoded:

```
encode ? encodeSpecialCharacters(title) : title
```